src/pages/UploadValidator/index.tsx

```diff
@@ -128,11 +178,23 @@ const _UploadValidatorPage = ({
               setIsFileAccepted(false);
               dispatchDepositFileKeyUpdate([]);
               flushDropzoneCache();
+
+              // there are a couple special cases that can occur
+              const { fork_version: forkVersion } = fileData[0] || {};
+              const hasCorrectStructure = checkJsonStructure(fileData[0] || {});
+              if (
+                hasCorrectStructure &&
+                forkVersion !== (GENESIS_FORK_VERSION as string)
+              ) {
+                // file doesn't match the correct network
+                handleWrongNetwork();
+              }
             }
           } catch (e) {
             // possible error example: json is invalid or empty so it cannot be parsed
             // TODO think about other possible errors here, and consider if we might want to set "isFileStaged"
             setIsFileAccepted(false);
+            handleSevereError();
```

Comment:
To be particular, this is the only time the scary error is shown.

Reply:
Indeed, but uploading `keystore.json` instead of `deposit_data.json` when they are both in the same directory is a very easy thing to do and then the user sees the Big Scary.

Reply:
I think I’m misunderstanding you since that scenario wouldn’t wouldn’t reach the catch block. My apologies for that (insomnia, maaaaan. haha)

Are you saying that somebody _should_ get the scary error if they try uploading their `keyfile.json`?

Reply:
Would it be better to show a different scary message than the BigScary™️ one that’s shown now that conveys something like

> “Dude. This is your keyfile, not the deposit_data. You should be guarding this with your life. Try again.”

?

Reply:
> (insomnia, maaaaan. haha)

Can relate. I actually think I checked out the wrong version of this branch. I tried it again and I think it is fine, apologies.